llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -749,10 +749,9 @@ void AArch64AsmPrinter::emitHwasanMemaccessSymbols(Module &M) {
 
   const Triple &TT = TM.getTargetTriple();
   assert(TT.isOSBinFormatELF());
-  std::unique_ptr<MCSubtargetInfo> STI(
-      TM.getTarget().createMCSubtargetInfo(TT, "", ""));
-  assert(STI && "Unable to create subtarget info");
-  this->STI = static_cast<const AArch64Subtarget *>(&*STI);
+  AArch64Subtarget STI(TT, TM.getTargetCPU(), TM.getTargetCPU(),
```

Comment:
Won't these strings be set to arbitrary values, like the last function in the module or something? The reason I used `""` was that we don't have a function to get cpu/features from in both of these cases.

Reply:
They shouldn't be. Although the various `getSubtargetImpl(const Function &)` functions for different targets do call `resetTargetOptions(F)`, that only sets a few specific options on the `TargetMachine` level. For CPU and target features, they create local variables for that and leave the `TargetMachine`'s fields alone.

Reply:
I see, it looks like these subtarget fields on the TargetMachine are intended to be used for things like module-level inline asm. This sounds fine to me then because that's more or less what we are generating here. I guess we'll end up having to do something else if we ever develop a concept of subtargets for module-level inline asm, but this is okay for now.